src/traces/scattermapbox/attributes.js

```diff
@@ -26,6 +26,61 @@ module.exports = overrideAll({
     lon: scatterGeoAttrs.lon,
     lat: scatterGeoAttrs.lat,
 
+    cluster: {
+        enabled: {
+            valType: 'boolean',
+            role: 'info',
+            dflt: false,
+            editType: 'plot',
+            description: 'Determines whether clustering is enabled or disabled.'
+        },
+        maxzoom: {
+            valType: 'number',
+            role: 'info',
+            min: 0,
+            max: 22,
+            dflt: 14,
+            description: 'Sets the maximum zoom level for the cluster.',
+        },
+        radius: {
+            role: 'info',
+            valType: 'number',
+            dflt: 50,
+            description: 'Radius of each cluster when clustering points.',
+        },
+        steps: {
+            role: 'info',
+            valType: 'number',
+            arrayOk: true,
+            dflt: -1,
+            min: -1,
+            description: [
+                'Sets the steps for each cluster.'
+            ].join(' ')
+        },
+        color: {
```

Comment:
It seems we could drop `color` and `size` here and reuse `trace.marker.color` and `trace.marker.size`.

Reply:
I don't think we can drop color and size. Shouldn't one be able to have different color and sized point and clusters. If we use the color from `trace.marker.color` it will no be possible to have a cluster with say 3 different colors, and the points being pained with another color

Reply:
I think instead of having color and size both be arrays, we might want to have a single attribute that accepts an array of color/size tuples like `steps: [ {color: "red", size: 10} ]` or something?

Reply:
So we need to have either a `color`/`size` to be a singular value or an array. If `color`/`size` is an array, steps should also be defined as an array that has length one less than `color`/`size`. `steps` defines at which levels the cluster changes `color`/`size` dependent on how many observations the cluster contains. 

One way to achieve that would be

`steps: [ {color: "red", size: 10, "step": null}, {color: "blue", size: 20, "step": 20} ]`

In this case, the cluster will be coloured red and sized 10 as long as there are less than 20 observations in the cluster else the cluster will be coloured blue and sized 20.